### ISSUE TITLE
User Settings: Implement setting-provided default values

### ISSUE THREAD

**Issue by tjcouch-sil** [paranext/paranext-core#709]
As an extension developer, I want to be able to get settings' values including defaults like interface language so I can stay in sync whether or not the setting has been set.

----

On the settings service:
- Maintain list of all user settings' default values (hard-coded; we will load from extensions in #719)
  - Include some real settings - interface language, current Scripture reference
- Use the default value from the manifest instead of the parameter default value passed in. No more `defaultValue` parameter in service or in hook.
  - Rework settings service's data provider: `getSetting(key)`
    - If setting is provided in extensions or built into core (though for now the list is just built into core as mentioned above), returns setting value or its default if it doesn't have a value
    - If setting is not provided anywhere, returns setting value if it has one or throws if it doesn't have a value
      - Should we revert this and say `defaultValue` parameter can be a backup for the extension-provided default? I think it would be nice never to have to worry about if a setting exists
      - YAGNI. We can do something like this if we need it one day for React stuff. Extensions should define their own settings and depend on extensions whose settings they absolutely need. We can figure something out for optional dependencies if we want. Maybe add a new `papi.settings.tryGet` or something that returns an object with success and the value

Note: intentionally not adding `hasSetting(key)` because the use case isn't clear and it could cause race condition issues. Just try/catch `getSetting` if you're unsure if a setting is available until we find a very good reason to do otherwise.

Depends on #502

**Comment by tjcouch-sil** [comment]
Closed - merged into #502 as this is trivially more work, and it would be a waste of time to create the other api then remove it to add this one.
